### PR TITLE
CB-11958 Gcp database and compute resource checkers retry on socket and connect timeout exceptions

### PR DESCRIPTION
**Pull request by biharitomi**
See detailed description in the commit message.

### CODE REVIEW

**Review by doktoric — APPROVED** [review]

File: cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/service/checker/GcpDatabaseResourceChecker.java [hunk 1]
```diff
@@ -36,13 +49,26 @@ protected void checkSqlOperationError(com.google.api.services.sqladmin.model.Ope
...
                 msg = error.toString();
+                LOGGER.warn("Gcp database operation failed and return with the following errors: '{}'", msg);
             }
             throw new CloudConnectorException(msg);
         }
     }
+
+    @VisibleForTesting
+    SQLAdmin.Operations.Get getSqlAdminOperations(SQLAdmin sqlAdmin, String operationId, String projectId) throws IOException {
+        return GcpStackUtil.sqlAdminOperations(sqlAdmin, projectId, operationId);
```
Comment (topolyai5) [inline cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/service/checker/GcpDatabaseResourceChecker.java:66]: maybe better to write the test for the static method

File: cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/service/checker/GcpDatabaseResourceChecker.java [hunk 2]
```diff
@@ -36,13 +49,26 @@ protected void checkSqlOperationError(com.google.api.services.sqladmin.model.Ope
...
+    }
+
+    @VisibleForTesting
+    @NotNull
+    String getProjectId(AuthenticatedContext ac) {
+        return GcpStackUtil.getProjectId(ac.getCloudCredential());
```
Comment (topolyai5) [inline cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/service/checker/GcpDatabaseResourceChecker.java:72]: maybe better to write the test for the static method

File: cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/service/checker/GcpDatabaseResourceChecker.java [hunk 3]
```diff
@@ -36,13 +49,26 @@ protected void checkSqlOperationError(com.google.api.services.sqladmin.model.Ope
...
         }
     }
+
+    @VisibleForTesting
+    SQLAdmin.Operations.Get getSqlAdminOperations(SQLAdmin sqlAdmin, String operationId, String projectId) throws IOException {
+        return GcpStackUtil.sqlAdminOperations(sqlAdmin, projectId, operationId);
+    }
+
+    @VisibleForTesting
+    @NotNull
```
Comment (bergerdenes) [inline cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/service/checker/GcpDatabaseResourceChecker.java:70]: please remove JetBrains annotation

**Review by bergerdenes — CHANGES_REQUESTED** [review]

File: cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/service/checker/GcpDatabaseResourceChecker.java (hunk 1 above)
Comment (biharitomi) [reply cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/service/checker/GcpDatabaseResourceChecker.java:66]: I like this approach 😄

Comment (topolyai5) [reply cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/service/checker/GcpDatabaseResourceChecker.java:66]: I mean, write the test for this class: `GcpStackUtil`. You don't mock the static method neither here nor there

Comment (topolyai5) [reply cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/service/checker/GcpDatabaseResourceChecker.java:66]: and you don't need to set the modifier to package protected if you move the test to `GcpStackUtilTest`

**Review by bergerdenes — APPROVED** [review]

**Review by topolyai5 — APPROVED** [review]

**Pull request merged**